jujugui/static/gui/src/app/components/deployment-flow/sshkey/sshkey.js

```diff
@@ -30,15 +30,310 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 const AZURE_CLOUD_TYPE = 'azure';
 
 class DeploymentSSHKey extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      addSource: 'github',
+      SSHkeys: [],
+      error: null,
+      buttonDisabled: true
+    };
+  }
+
+  /** Handle keyboard input to listen for return.
+
+    @method _onSSHKeyInputKey
+    @param {Object} evt The keyboard event.
+  */
+  _onSSHKeyInputKey(evt) {
+    if (evt.which === 13) {
+      this._handleAddMoreKeys(null);
+    }
+    this.setState({
+      buttonDisabled: this.refs.sshKey.getValue() ? false : true
+    });
+  }
+
+  /** Handle keyboard input to listen for return.
+
+    @method _onSSHKeyInputKey
+    @param {Object} evt The keyboard event.
+  */
+  _onGithubUsernameInputKey(evt) {
+    if (evt.which === 13) {
+      this._handleAddMoreKeys(null);
+    }
+    this.setState({
+      buttonDisabled: this.refs.githubUsername.getValue() ? false : true
+    });
+  }
+
+  /**
+    Split a Manual Key into its parts
+
+    @method _splitKey
+    @param {String} sshkey
+   */
+  _splitKey(sshKey) {
+    const splitKey = sshKey.split(' ');
+    return {
+      'body': splitKey[1],
+      'text': sshKey,
+      'type': splitKey[0],
+      'id': 0
+    };
+  }
+
+  /**
+    Check if an sshkey exists to avoid duplication.
+
+    @method _keyExists
+    @param {Object} key The key object to check.
+    @return {Boolean}
+  */
+  _keyExists(key) {
+    return this.state.SSHkeys.filter(knownKey => {
```

Comment:
Take it or leave it: `this.state.SSHkeys.findIndex(knownKey => knownKey.body === key.body) !== -1;`